### PR TITLE
Fix derive Data when type param bounds are defined

### PR DESCRIPTION
**Pull request by chris-zen**
This PR addresses #1034

### CODE REVIEW

File: druid/src/data.rs [hunk 1]
```diff
@@ -474,3 +474,34 @@ mod test {
         assert!(!one.same(&two));
     }
 }
+
+#[cfg(test)]
+mod tests {
```
Comment (Finnerale) [inline druid/src/data.rs:479]: I feel like those belong either into `druid-derive/tests/data.rs` or a new `druid-derive/tests/generics.rs`.

Comment (chris-zen) [reply druid/src/data.rs]: Good points. Thanks. Done.

**Comment by chris-zen** [comment]
@xStrom @Finnerale I don't know why clipper+test are failing to compile `piet-coregraphics`.

**Comment by cmyr** [comment]
it's an upstream issue, see https://github.com/linebender/piet/pull/220

**Comment by jneem** [comment]
I think the upstream issue is fixed, so I'm re-running the job.

File: CHANGELOG.md [hunk 2]
```diff
@@ -26,6 +26,7 @@ You can find its changes [documented below](#060---2020-06-01).
 - macOS: Timers not firing during modal loop. ([#1028] by [@xStrom])
 - GTK: Directory selection now properly ignores file filters. ([#957] by [@xStrom])
 - GTK: Don't crash when receiving an external command while a file dialog is visible. ([#1043] by [@jneem])
+- Fix derive `Data` when type param bounds are defined ([#1058] by [@chris-zen])
```
Comment (Finnerale) [inline CHANGELOG.md:29]: Sadly GitHub does not automatically link those, the links must be added manually at the end of the file.

File: CHANGELOG.md [hunk 3]
```diff
@@ -27,7 +27,9 @@ You can find its changes [documented below](#060---2020-06-01).
...
 - Ensure that `update` is called after all commands. ([#1062] by [@jneem])
+- Fix derive `Data` when type param bounds are defined ([#1058] by [@chris-zen])
```
Comment (Finnerale) [inline CHANGELOG.md:32]: Now we have the entry twice :D

Comment (chris-zen) [reply CHANGELOG.md]: :man_facepalming: sorry, I messed up while merging.

**Pull request merged**